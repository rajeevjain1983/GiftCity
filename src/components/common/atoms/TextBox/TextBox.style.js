// 9fbef606107a605d69c0edbcd8029e5d 
import { css } from 'styled-components';

const textboxStyles = css`
  position: relative;
  display: block;

  .TextBox__label {
    font-size: ${props =>
      props.input && props.input.value
        ? props.theme.fonts.fontSize.body.small.primary
        : props.theme.fonts.fontSize.textbox}px;
    padding: 0;
    position: absolute;
    top: ${props =>
      props.input && props.input.value
        ? props.theme.spacing.ELEM_SPACING.XXS
        : props.theme.spacing.ELEM_SPACING.LRG};
    ${props =>
      props.input &&
      props.input.value &&
      `
      font-weight: ${props.theme.fonts.fontWeight.bold};
    `}
  }
  .TextBox__view {
    font-size: 16px;
    position: absolute;
    top: 10px;
    padding: 5px;
    color: ${props => props.theme.colors.TEXTBOX.COLOR};
  }
  .TextBox__input {
    margin: 0;
    outline: 0;
    font-size: ${props => props.theme.fonts.fontSize.textbox_input}px;
    color: ${props => props.theme.colors.TEXTBOX.COLOR};
    width: 100%;
    background-position: left top;
    background-repeat: no-repeat;
    background-size: contain;
    border: 0 solid transparent;
    border-bottom: 1px solid ${props => props.theme.colors.FOOTER.DIVIDER};
    padding-bottom: ${props => props.theme.spacing.ELEM_SPACING.XS};
    margin-bottom: ${props => props.theme.spacing.ELEM_SPACING.XS};
    padding-top: ${props => props.theme.spacing.ELEM_SPACING.LRG};

    ${props =>
      props.meta &&
      props.meta.touched &&
      props.meta.error &&
      `
      border-bottom: 1px solid ${props.theme.colors.NOTIFICATION.ERROR};
    `}

    ${props =>
      props.disabled
        ? `
      background-color: ${props.theme.fieldBackgroundDisabledColor};
      border-color: ${props.theme.fieldBorderDisabledColor};
    `
        : ''};

    &:focus + .TextBox__label {
      font-size: ${props => props.theme.fonts.fontSize.body.small.primary}px;
      font-weight: ${props => props.theme.fonts.fontWeight.bold};
      top: ${props => props.theme.spacing.ELEM_SPACING.XXS};
    }
  }
  .success__checkmark {
    display: none;
  }
  &.textbox_validation_success .TextBox__input {
    border-bottom: 1px solid ${props => props.theme.colors.success};
  }

  &.textbox_validation_success .success__checkmark {
    display: ${props => (props.input && props.type === 'hidden' ? 'none' : 'block')};
    width: 15px;
    height: 8px;
    margin-left: ${props => props.theme.spacing.ELEM_SPACING.XXXS};
    transform: rotate(-45deg);
    transform-origin: left;
    position: absolute;
    right: ${props => (props.checkMarkRightMargin ? props.checkMarkRightMargin : 0)};
    top: 30px;

    &:before {
      content: '';
      position: absolute;
      width: 3px;
      height: 100%;
      background-color: ${props => props.theme.colors.TEXTBOX.SUCCESS_BORDER};
    }

    &:after {
      content: '';
      position: absolute;
      width: 100%;
      height: 3px;
      background-color: ${props => props.theme.colors.TEXTBOX.SUCCESS_BORDER};
      bottom: 0;
    }
  }

  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};

  .TextBox__error {
    display: flex;
    flex-direction: row;
    padding-bottom: ${props => props.theme.spacing.ELEM_SPACING.XXS};
  }
  .warning-icon {
    /* background: transparent url(${getIconPath('circle-alert-fill')}) no-repeat 0 0; */
    background-size: contain;
    border: none;
    height: 14px;
    width: 16px;
    margin-right: 7px;
    flex-shrink: 0;
  }

  .TextBox_view {
    border: 1px solid gray;
    border-radius: 4px;
    height: 140px;
    width: 100%;
    margin-bottom: 10px;
  }
  .TextBox_view_border {
    border: 1px solid red;
    border-radius: 4px;
    height: 140px;
    width: 100%;
    margin-bottom: 10px;
  }
`;

export default textboxStyles;
