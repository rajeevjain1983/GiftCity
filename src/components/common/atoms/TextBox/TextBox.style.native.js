// 9fbef606107a605d69c0edbcd8029e5d 
import styled, { css } from 'styled-components/native';

const StyledTextBoxWrapper = styled.View`
  margin-bottom: ${props => (props.marginBottom ? props.theme.spacing.ELEM_SPACING.XL : 0)};
`;

const getInputBottomColor = props => {
  const { theme, meta } = props;
  const { colorPalette } = theme;
  const { dirty, error } = meta;
  const borderColor = dirty && error !== undefined ? colorPalette.error : colorPalette.gray[1500];
  return `
  border-bottom-color: ${borderColor};
  `;
};

const TextBoxStyle = css`
  ${getInputBottomColor};
`;

const StyledTextBox = styled.TextInput`
  border-bottom-width: 1px;
  height: ${props => (props.isCVVField ? '28px' : '40px')};
  padding-top: ${props => (props.isCVVField ? '0' : props.theme.spacing.ELEM_SPACING.MED)};
  padding-bottom: 0;
  padding-left: 0;
  font-family: ${props => props.theme.typography.fonts.secondary};
  font-weight: ${props => props.theme.typography.fontWeights.regular};
  border-bottom-color: ${props =>
    props.meta.touched && props.meta.error
      ? props.theme.colorPalette.error
      : props.theme.colorPalette.gray[600]};
  color: ${props => props.theme.colorPalette.gray[900]};
`;

const StyledLabel = styled.Text`
  position: absolute;
  left: 0;
  top: ${props => (!props.isFocused ? props.theme.spacing.ELEM_SPACING.MED : '0')};
  font-family: ${props => props.theme.typography.fonts.secondary};
  font-size: ${props =>
    !props.isFocused
      ? props.theme.typography.fontSizes.fs14
      : props.theme.typography.fontSizes.fs10};
  color: #1a1a1a;
  font-weight: ${props =>
    !props.isFocused
      ? props.theme.typography.fontWeights.regular
      : props.theme.typography.fontWeights.extrabold};
  margin-bottom: ${props => (props.isFocused ? props.theme.spacing.ELEM_SPACING.XXS : '0')};
`;

const StyledErrorIcon = styled.View`
  margin-right: ${props => props.theme.spacing.ELEM_SPACING.XS};
`;

const StyledSuccessCheck = styled.View`
  margin-right: ${props => props.theme.spacing.ELEM_SPACING.XXS};
`;

const StyledErrorWrapper = styled.View`
  margin-top: ${props => props.theme.spacing.ELEM_SPACING.SM};
  display: flex;
  flex-direction: row;
  align-items: ${props => (props.isCVVFieldError ? 'stretch' : 'center')};
  width: 90%;
  ${props => (props.errorStyle ? props.errorStyle : '')}
`;

const StyledSuccessIcon = styled.View`
  position: absolute;
  right: 0;
  top: ${props => props.theme.spacing.ELEM_SPACING.LRG};
`;

const HiddenView = styled.View`
  display: none;
`;

export {
  TextBoxStyle,
  StyledTextBox,
  StyledLabel,
  StyledErrorIcon,
  StyledErrorWrapper,
  StyledTextBoxWrapper,
  StyledSuccessIcon,
  HiddenView,
  StyledSuccessCheck,
};
