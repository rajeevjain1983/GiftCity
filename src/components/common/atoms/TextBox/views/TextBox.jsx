// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import BodyCopy from '../../BodyCopy';
import withStyles from '../../../hoc/withStyles';
import StyledTextBox from '../TextBox.style';

/**
 *
 * @param {Boolean} showSuccessCheck Forcefully show the success mark and success border. Override redux form params.
 * @param {Object} meta redux-form meta object to analyse the success state. Based on this class will be returned.
 * @return {string} Returns a class name if the form success validation criteria matches else blank string.
 */
const getValidationSuccessClass = (enableSuccessCheck, showSuccessCheck, meta) => {
  const { invalid, pristine, asyncValidating, active } = meta;
  return enableSuccessCheck &&
    (showSuccessCheck || (!active && !pristine && !invalid && !asyncValidating))
    ? 'textbox_validation_success'
    : '';
};

const TextBox = ({
  className,
  id,
  ariaLabel,
  type,
  placeholder,
  maxLength,
  input,
  inputRef,
  meta,
  dataLocator,
  showSuccessCheck,
  enableSuccessCheck,
  isRequired,
  errorDataLocator,
  submitError,
  showExplicitError,
  roundTextBox,
  ...others
}) => {
  const elemValue = input.value;
  const { touched, error = submitError } = meta;
  const errorMessagea11yLbl = `textbox__error__${input.name}`;
  const isErrored = (touched && error) || showExplicitError;

  return (
    <label
      htmlFor={id}
      className={`${className} input-fields-wrapper ${getValidationSuccessClass(
        enableSuccessCheck,
        showSuccessCheck,
        meta
      )}`}
    >
      <input
        {...others}
        {...input}
        id={id}
        aria-label={ariaLabel || placeholder}
        className={roundTextBox ? 'TextBox_view' : 'TextBox__input'}
        name={input.name}
        type={type}
        maxLength={maxLength}
        value={elemValue}
        ref={inputRef}
        placeholder=""
        data-locator={dataLocator}
        aria-required={isRequired}
        aria-describedby={errorMessagea11yLbl}
      />
      <BodyCopy
        className={roundTextBox ? 'TextBox__view' : 'TextBox__label'}
        fontFamily="secondary"
        fontSize="fs12"
      >
        {placeholder}
      </BodyCopy>
      <div className="TextBox__error">
        <div className={isErrored ? 'warning-icon' : ''} aria-disabled="true" />
        <BodyCopy
          color="error"
          component="div"
          fontSize="fs12"
          fontFamily="secondary"
          fontWeight="extrabold"
          role="alert"
          aria-live="assertive"
          data-locator={errorDataLocator}
          id={errorMessagea11yLbl}
        >
          {isErrored ? error || showExplicitError : ''}
        </BodyCopy>
      </div>
      {enableSuccessCheck && <div className="success__checkmark" />}
    </label>
  );
};

TextBox.defaultProps = {
  id: '',
  ariaLabel: '',
  type: 'text',
  placeholder: '',
  errorDataLocator: '',
  onChangeHandler: () => {},
  dataLocator: '',
  meta: {},
  showSuccessCheck: false,
  enableSuccessCheck: true,
  isRequired: false,
  submitError: '',
  showExplicitError: '',
  roundTextBox: false,
};

TextBox.propTypes = {
  className: PropTypes.string.isRequired,
  id: PropTypes.string,
  ariaLabel: PropTypes.string,
  placeholder: PropTypes.string,
  maxLength: PropTypes.number.isRequired,
  inputRef: PropTypes.node.isRequired,
  meta: PropTypes.shape({}),
  dataLocator: PropTypes.string,
  enableSuccessCheck: PropTypes.bool,
  isRequired: PropTypes.bool,
  errorDataLocator: PropTypes.string,
  input: PropTypes.shape({}).isRequired,
  type: PropTypes.string,
  showSuccessCheck: PropTypes.bool,
  submitError: PropTypes.string,
  showExplicitError: PropTypes.string,
  onChangeHandler: PropTypes.func,
  roundTextBox: PropTypes.bool,
};
export default withStyles(TextBox, StyledTextBox);
export { TextBox as TextBoxVanilla };
