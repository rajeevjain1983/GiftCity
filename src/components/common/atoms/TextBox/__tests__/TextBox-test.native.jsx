// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { TextBox } from '../views/TextBox.native';

describe('Textbox component', () => {
  const props = {
    id: 'input',
    ariaLabel: 'input',
    type: 'text',
    meta: { touched: '', error: '' },
    dataLocator: 'input-field',
    enableSuccessCheck: false,
    label: 'input',
    keyboardType: 'default',
    showErrorIcon: true,
    input: { value: 'foo' },
  };

  let component;

  beforeEach(() => {
    component = shallow(<TextBox {...props} />);
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders correctly hidden textbox', () => {
    component.setProps({ type: 'hidden' });
    expect(component).toMatchSnapshot();
  });

  it('renders correctly explicit error with textbox', () => {
    component.setProps({ type: 'text', showExplicitError: 'Some Error Occurred' });
    expect(component).toMatchSnapshot();
  });

  it('render enableSuccessCheck on validation true', () => {
    component.setProps({ enableSuccessCheck: true });
    expect(component).toMatchSnapshot();
  });

  it('render enableSuccessCheck on validation fail', () => {
    component.setProps({ meta: { touched: 'foo', error: 'food' } });
    expect(component).toMatchSnapshot();
  });

  it('check handleFocus', () => {
    component.setState({ isFocused: false });
    component.instance().handleFocus();
    expect(component.state('isFocused')).toBe(true);
  });

  it('check handleBlur', () => {
    component.setProps({ input: { onBlur: jest.fn(), value: '' } });
    component.setState({ isFocused: true });
    component.instance().handleBlur();
    expect(component.state('isFocused')).toBe(false);
  });
});
