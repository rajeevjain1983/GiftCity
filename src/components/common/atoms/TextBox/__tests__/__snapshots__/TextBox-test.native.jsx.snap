// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox component render enableSuccessCheck on validation fail 1`] = `
<View>
  <View>
    <View>
      <Styled(Text)
        isFocused="foo"
      >
        input
      </Styled(Text)>
      <Styled(Component)
        accessibilityLabel="input"
        ariaLabel="input"
        autoCapitalize="sentences"
        className="TextBox__input"
        data-locator="input-field"
        dataLocator="input-field"
        enableSuccessCheck={false}
        error="food"
        id="input"
        isCVVFieldError={false}
        keyboardType="default"
        label="input"
        meta={
          Object {
            "error": "food",
            "touched": "foo",
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onPasswordFocus={false}
        returnKeyType="next"
        secureTextEntry={false}
        showSuccessCheck={false}
        successText=""
        type="text"
        value="foo"
      />
    </View>
  </View>
</View>
`;

exports[`Textbox component render enableSuccessCheck on validation true 1`] = `
<View>
  <View>
    <View>
      <Styled(Text)
        isFocused="foo"
      >
        input
      </Styled(Text)>
      <Styled(Component)
        accessibilityLabel="input"
        ariaLabel="input"
        autoCapitalize="sentences"
        className="TextBox__input"
        data-locator="input-field"
        dataLocator="input-field"
        enableSuccessCheck={true}
        error=""
        id="input"
        isCVVFieldError={false}
        keyboardType="default"
        label="input"
        meta={
          Object {
            "error": "",
            "touched": "",
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onPasswordFocus={false}
        returnKeyType="next"
        secureTextEntry={false}
        showSuccessCheck={false}
        successText=""
        type="text"
        value="foo"
      />
      <Styled(View)>
        <Styled(ImageComp)
          alt=""
          cacheType="immutable"
          height="12px"
          host=""
          isFastImage={false}
          onError={[Function]}
          onLoadStart={[Function]}
          onProgress={[Function]}
          priority="normal"
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/images/success-icon.png",
            }
          }
          url=""
          width="15px"
        />
      </Styled(View)>
    </View>
    <Styled(View)
      marginBottom={true}
    />
  </View>
</View>
`;

exports[`Textbox component renders correctly 1`] = `
<View>
  <View>
    <View>
      <Styled(Text)
        isFocused="foo"
      >
        input
      </Styled(Text)>
      <Styled(Component)
        accessibilityLabel="input"
        ariaLabel="input"
        autoCapitalize="sentences"
        className="TextBox__input"
        data-locator="input-field"
        dataLocator="input-field"
        enableSuccessCheck={false}
        error=""
        id="input"
        isCVVFieldError={false}
        keyboardType="default"
        label="input"
        meta={
          Object {
            "error": "",
            "touched": "",
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onPasswordFocus={false}
        returnKeyType="next"
        secureTextEntry={false}
        showSuccessCheck={false}
        successText=""
        type="text"
        value="foo"
      />
    </View>
    <Styled(View)
      marginBottom={true}
    />
  </View>
</View>
`;

exports[`Textbox component renders correctly explicit error with textbox 1`] = `
<View>
  <View>
    <View>
      <Styled(Text)
        isFocused="foo"
      >
        input
      </Styled(Text)>
      <Styled(Component)
        accessibilityLabel="input"
        ariaLabel="input"
        autoCapitalize="sentences"
        className="TextBox__input"
        data-locator="input-field"
        dataLocator="input-field"
        enableSuccessCheck={false}
        error=""
        id="input"
        isCVVFieldError={false}
        keyboardType="default"
        label="input"
        meta={
          Object {
            "error": "",
            "touched": "",
          }
        }
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        onFocus={[Function]}
        onPasswordFocus={false}
        returnKeyType="next"
        secureTextEntry={false}
        showSuccessCheck={false}
        successText=""
        type="text"
        value="foo"
      />
    </View>
    <Styled(View)
      marginBottom={true}
    >
      <Styled(View)
        isCVVFieldError={false}
      >
        <Styled(View)>
          <Styled(ImageComp)
            alt=""
            cacheType="immutable"
            height="14px"
            host=""
            isFastImage={false}
            onError={[Function]}
            onLoadStart={[Function]}
            onProgress={[Function]}
            priority="normal"
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/alert-triangle.png",
              }
            }
            url=""
            width="16px"
          />
        </Styled(View)>
        <Styled(BodyCopy)
          accessibilityText=""
          color="error"
          dataLocator=""
          fontFamily="secondary"
          fontSize="fs12"
          fontWeight="extrabold"
          margin={null}
          numberOfLines={null}
          text="Some Error Occurred"
          textDecoration={null}
        />
      </Styled(View)>
    </Styled(View)>
    <Styled(View)
      marginBottom={true}
    />
  </View>
</View>
`;

exports[`Textbox component renders correctly hidden textbox 1`] = `
<View>
  <View>
    <Styled(View)>
      <View>
        <Styled(Text)
          isFocused="foo"
        >
          input
        </Styled(Text)>
        <Styled(Component)
          accessibilityLabel="input"
          ariaLabel="input"
          autoCapitalize="sentences"
          className="TextBox__input"
          data-locator="input-field"
          dataLocator="input-field"
          enableSuccessCheck={false}
          error=""
          id="input"
          isCVVFieldError={false}
          keyboardType="default"
          label="input"
          meta={
            Object {
              "error": "",
              "touched": "",
            }
          }
          onBlur={[Function]}
          onChangeText={[Function]}
          onEndEditing={[Function]}
          onFocus={[Function]}
          onPasswordFocus={false}
          returnKeyType="next"
          secureTextEntry={false}
          showSuccessCheck={false}
          successText=""
          type="hidden"
          value="foo"
        />
      </View>
    </Styled(View)>
  </View>
</View>
`;
